Support Fixed Server ID Encoded Load Balancing Mode (#3172)
microsoft/msquic: docs/Deployment.md

+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
```

`Server ID` is 4 bytes long, and can encode various data interpretations.

This encoding is **not enabled by default**. Please set the `LoadBalancingMode` setting accordingly to configure the desired load balancing mode. By default, <todo>.

> **Note**

> > Any time the `LoadBalancingMode` is updates, the server must be restarted for this setting to take effect.

Reviewer: ```suggestion
> Any time the `LoadBalancingMode` is updated, the server must be restarted for this setting to take effect.
```

## IP Address Encoded Server ID

> MsQuic supports encoding the local IPv4 address or IPv6 suffix into Server ID. With this load balancing model, the load balancer can use this encoded IP address/suffix to directly route received QUIC packets to the appropriate server.

Reviewer: ```suggestion
MsQuic supports encoding the local IPv4 address or IPv6 suffix into Server ID. The load balancer can use this encoded IP address/suffix to directly route received QUIC packets to the appropriate server.
```

To configure this mode, set the `LoadBalancingMode` setting to `1`.

## Fixed Server ID

> MsQuic supports encoding a fixed, 4-byte value into Server ID. With this load balancing model, the load balancer can use this value to lookup and route received QUIC packets to the appropriate server.

Reviewer: ```suggestion
MsQuic supports encoding a fixed, 4-byte value into Server ID. The load balancer can use this value to look up and route received QUIC packets to the appropriate server.
```

> To configure this mode, set the `LoadBalancingMode` setting to `2` and the `FixedServerID` setting to your desired value.

Reviewer 2: Are these regkeys then? or are apps using `SetParam`?

Author: It's an API setting that is also backed by a registry key.
